### ISSUE TITLE
Ruby API

### ISSUE THREAD

**Issue by FabianShallari** [tensorflow/tensorflow#50]
`we’re hoping to entice you to contribute SWIG interfaces to your favorite language -- be it Go, Java, Lua, Javascript, or R.` - no Ruby, really? c'mon. Ruby is awesome

**Comment by marcoceppi** [comment]
I don't think that was meant to be an exhaustive list...

**Comment by mensfeld** [comment]
+1

**Comment by vincentvanhoucke** [comment]
As you have rightfully guessed, the core TensorFlow team doesn't have plans for Ruby on the horizon, so if you want to contribute it, please go ahead! Submitting drafts for frontends such as this on the discuss mailing list early on is probably a good idea.

**Comment by babakgh** [comment]
:+1:

**Comment by ManuelArno** [comment]
I will pay for a Ruby api.

2015-11-12 11:39 GMT+01:00 Babak Gh <notifications@github.com>:

> [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-156067031>
> .
>



-- 
Manuel Arno Korfmann
Luetzowstrasse 8
24105 Kiel

manu @ korfmann . info
+49 40 210913788

www.manuel-korfmann.de

**Comment by smackesey** [comment]
+1. I am willing to work on this, it would be good to coordinate with anyone else interested.

**Comment by albertog2man** [comment]
+1 Let me know of any efforts. I will for sure be forking that.

**Comment by arbirk** [comment]
+1

**Comment by ManuelArno** [comment]
me too! if help is need just pink manu@korfmann.info

2015-11-14 15:04 GMT+00:00 arbirk <notifications@github.com>:

> +1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-156709440>
> .
>



-- 
Manuel Arno Korfmann
Luetzowstrasse 8
24105 Kiel

manu @ korfmann . info
+49 40 210913788

www.manuel-korfmann.de

**Comment by ManuelArno** [comment]
*ping

2015-11-14 15:06 GMT+00:00 Manuel Korfmann <manu@korfmann.info>:

> me too! if help is need just pink manu@korfmann.info
>
> 2015-11-14 15:04 GMT+00:00 arbirk <notifications@github.com>:
>
>> +1
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-156709440>
>> .
>>
>
>
>
> --
> Manuel Arno Korfmann
> Luetzowstrasse 8
> 24105 Kiel
>
> manu @ korfmann . info
> +49 40 210913788
>
> www.manuel-korfmann.de
>
>
>


-- 
Manuel Arno Korfmann
Luetzowstrasse 8
24105 Kiel

manu @ korfmann . info
+49 40 210913788

www.manuel-korfmann.de

**Comment by nicolas-fricke** [comment]
+1 I'd be interested in contributing, too! :)

**Comment by M-Medhat** [comment]
+1

**Comment by ggomagundan** [comment]
+1

**Comment by devhyunjae** [comment]
+12

**Comment by scsilver** [comment]
+1

**Comment by ManuelArno** [comment]
If you guys want to do RNN you can checkout @josephwilk's tlearn.rb.

2015-11-30 2:50 GMT+01:00 scsilver <notifications@github.com>:

> +1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-160496474>
> .
>



-- 
Manuel Arno Korfmann
Luetzowstrasse 8
24105 Kiel

manu @ korfmann . info
+49 40 210913788

www.manuel-korfmann.de

**Comment by ManuelArno** [comment]
https://github.com/josephwilk/tlearn-rb

2015-11-30 8:49 GMT+01:00 Manuel Korfmann <manu@korfmann.info>:

> If you guys want to do RNN you can checkout @josephwilk's tlearn.rb.
>
> 2015-11-30 2:50 GMT+01:00 scsilver <notifications@github.com>:
>
>> +1
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-160496474>
>> .
>>
>
>
>
> --
> Manuel Arno Korfmann
> Luetzowstrasse 8
> 24105 Kiel
>
> manu @ korfmann . info
> +49 40 210913788
>
> www.manuel-korfmann.de
>
>
>


-- 
Manuel Arno Korfmann
Luetzowstrasse 8
24105 Kiel

manu @ korfmann . info
+49 40 210913788

www.manuel-korfmann.de

**Comment by flosstik** [comment]
+1

**Comment by jskDr** [comment]
I += 1

**Comment by arkiver** [comment]
+1

**Comment by jtoy** [comment]
+1

**Comment by zotherstupidguy** [comment]
+1

**Comment by jtoy** [comment]
Has anyone written a swig interface yet for tensorflow?  I would like to investigate writing this.

**Comment by chrisseldo** [comment]
+1

**Comment by soohanboys** [comment]
+ 1

**Comment by jskDr** [comment]
How's going on this subject?

Regards,
James
sungjinkim@fas.harvard.edu

Best regards,
Sungjin (James) Kim, Ph.D.
CCB in Harvard
sungjinkim@fas.harvard.edu

**Comment by nethsix** [comment]
+1. At your service.

**Comment by Neil-Aframe** [comment]
Interested, but is there a technical reason it needs to be SWIG? I can work in raw Ruby/C, Ruby/C++ fine, but find SWIG a bit too dense.

**Comment by jskDr** [comment]
How about using RICE?

**Comment by Fedcomp** [comment]
+1

**Comment by bver** [comment]
Great idea. 
Some clever numpy equivalent should be used for TF/Ruby with respect to interfacing TF tensors. 
https://github.com/SciRuby/nmatrix has an interface resembling numpy...?

**Comment by jtoy** [comment]
hi all, im very serious about getting this ported to ruby.  I do not have
the expertise to do that though.  I would be willing to fund  part (maybe
all) of this if someone or some team wants to work on it. Are any of you
interested in working on  this? I believe besides the  SWIG bindings
integration, we need the numpy like interface for manipulating the data.
My reasons for wanting this is simple. ruby has been cutoff from all the
scientific computing, but rubyists are great innovators and have great
ideas.  Tensorflow is a new software paradigm that is going to level the
playing field in deep learning and bring about a lot of new and wonderful
innovations. I want ruby community to be able to participate.  Please feel
free to email me,(check my profile) if interested in this.

**Comment by jskDr** [comment]
hi jtoy
I think NMatirx and SciRuby can be alternative tools for numpy. What do you
think?

On Fri, Mar 4, 2016 at 10:49 AM jtoy <notifications@github.com> wrote:

> hi all, im very serious about getting this ported to ruby. I do not have
> the expertise to do that though. I would be willing to fund part (maybe
> all) of this if someone or some team wants to work on it. Are any of you
> interested in working on this? I believe besides the SWIG bindings
> integration, we need the numpy like interface for manipulating the data.
> My reasons for wanting this is simple. ruby has been cutoff from all the
> scientific computing, but rubyists are great innovators and have great
> ideas. Tensorflow is a new software paradigm that is going to level the
> playing field in deep learning and bring about a lot of new and wonderful
> innovations. I want ruby community to be able to participate. Please feel
> free to email me,(check my profile) if interested in this.
>
> On Fri, Mar 4, 2016 at 10:00 AM, Pavel Suchmann <notifications@github.com>
> wrote:
>
> > Great idea.
> > Some clever numpy equivalent should be used for TF/Ruby with respect to
> > interfacing TF tensors.
> > https://github.com/SciRuby/nmatrix has an interface resembling numpy...?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/tensorflow/tensorflow/issues/50#issuecomment-192311660>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-192329422>
> .
>
-- 

Sungjin (James) Kim, PhD

Postdoc, CCB in Harvard

sungjinkim@fas.harvard.edu

[Web] http://aspuru.chem.harvard.edu/james-sungjin-kim/

[Linkedin] https://www.linkedin.com/in/jamessungjinkim

[Facebook] https://www.facebook.com/jamessungjin.kim

[alternative email] jamessungjin.kim@gmail.com

**Comment by jtoy** [comment]
I've seen both of those libraries and I know they have been around for a few years. I've never personally used them, but I know they are supposed to be alternatives to numpy/scipy. If we feel they can meet our needs, then I advocate that we use them instead of rewriting everything ourselves.

**Comment by nethsix** [comment]
This is going to happen in Google Summer of Code 2016, or/and probably earlier (not sure if GSoC requires students to be on board before starting).

**Comment by jskDr** [comment]
GS of Code 2016 is not a bad idea. We can try. Cooool~

On Fri, Mar 4, 2016 at 8:51 PM Neth Six <notifications@github.com> wrote:

> This is going to happen in Google Summer of Code 2016, or/and probably
> earlier (not sure if GSoC requires students to be on board before starting).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-192548691>
> .
>
-- 

Sungjin (James) Kim, PhD

Postdoc, CCB in Harvard

sungjinkim@fas.harvard.edu

[Web] http://aspuru.chem.harvard.edu/james-sungjin-kim/

[Linkedin] https://www.linkedin.com/in/jamessungjinkim

[Facebook] https://www.facebook.com/jamessungjin.kim

[alternative email] jamessungjin.kim@gmail.com

**Comment by jtoy** [comment]
it looks like an early stage proposal. we should contact the sciruby team to make this a reality then.

**Comment by nethsix** [comment]
Hey @jtoy . It is early stage. While we're awaiting applications from students to develop the Ruby tensorflow API, we're also researching into the best ways to do this. Our first steps would be to enable people to create a tensorflow graph using constants (scalar, matrices), and operations (add, multiply, etc.). Something along the lines of https://www.tensorflow.org/versions/r0.7/get_started/basic_usage.html#basic-usage

**Comment by JeffBusterCase** [comment]
I was waiting for this \*^\*. I want contribute too(I have a intermediate level of ruby programming skill) but  I believe I can do it(help)

**Comment by nethsix** [comment]
@JeffBusterCase Great! We can learn from one another and move this forward.

**Comment by riley-usagi** [comment]
What about now?

**Comment by nethsix** [comment]
Tensorflow Ruby API is one of the projects submitted to GSoC 2016 by
SciRuby. It is undergoing the selection process, and the results will be
out soon (22/23 Apr).

**Comment by jskDr** [comment]
Cool! I hope it would be accepted in oder that Ruby users can much enjoy
deep learning with the simple interfaced and powerful Tensorflow package. I
am now using it everyday for researching like medical drug discovery.
Especially I am using Tensorflow with Skflow which is renamed recently as
'learning' underneath of Tensorflow umbrella. So, I hope Ruby-Tensoflow may
include Skflow interface as well. What do you guys think?

On Sat, Apr 16, 2016 at 9:50 PM Neth Six <notifications@github.com> wrote:

> Tensorflow Ruby API is one of the projects submitted to GSoC 2016 by
> SciRuby. It is undergoing the selection process, and the results will be
> out soon (22/23 Apr).
>
> On Sun, Apr 17, 2016 at 2:18 AM, Riley Usagi <notifications@github.com>
> wrote:
>
> > What about now?
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/tensorflow/tensorflow/issues/50#issuecomment-210860945>
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-210937423>
>
-- 

Sungjin (James) Kim, PhD

Postdoc, CCB in Harvard

sungjinkim@fas.harvard.edu

[Web] http://aspuru.chem.harvard.edu/james-sungjin-kim/

[Linkedin] https://www.linkedin.com/in/jamessungjinkim

[Facebook] https://www.facebook.com/jamessungjin.kim

[alternative email] jamessungjin.kim@gmail.com

**Comment by nethsix** [comment]
Dr. Kim,

It's amazing to know that you are employing Tensorflow for real life
purposes!
You should write about it.
Using sklearn is an exercise in Udacity's Deep Learning course. I think it
helps people to get started easier.
Do help SciRuby develop this if you have time.

**Comment by jskDr** [comment]
Definitely. I am happy to help.
On Wed, Apr 20, 2016 at 9:26 PM Neth Six <notifications@github.com> wrote:

> Dr. Kim,
>
> It's amazing to know that you are employing Tensorflow for real life
> purposes!
> You should write about it.
> Using sklearn is an exercise in Udacity's Deep Learning course. I think it
> helps people to get started easier.
> Do help SciRuby develop this if you have time.
>
>
> On Thu, Apr 21, 2016 at 12:57 AM, Sungjin Kim <notifications@github.com>
> wrote:
>
> > Cool! I hope it would be accepted in oder that Ruby users can much enjoy
> > deep learning with the simple interfaced and powerful Tensorflow
> package. I
> > am now using it everyday for researching like medical drug discovery.
> > Especially I am using Tensorflow with Skflow which is renamed recently as
> > 'learning' underneath of Tensorflow umbrella. So, I hope Ruby-Tensoflow
> may
> > include Skflow interface as well. What do you guys think?
> >
> > On Sat, Apr 16, 2016 at 9:50 PM Neth Six <notifications@github.com>
> wrote:
> >
> > > Tensorflow Ruby API is one of the projects submitted to GSoC 2016 by
> > > SciRuby. It is undergoing the selection process, and the results will
> be
> > > out soon (22/23 Apr).
> > >
> > > On Sun, Apr 17, 2016 at 2:18 AM, Riley Usagi <notifications@github.com
> >
> > > wrote:
> > >
> > > > What about now?
> > > >
> > > > —
> > > > You are receiving this because you commented.
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
> https://github.com/tensorflow/tensorflow/issues/50#issuecomment-210860945>
> > > >
> > >
> > > —
> > > You are receiving this because you commented.
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/tensorflow/tensorflow/issues/50#issuecomment-210937423>
> > >
> > --
> >
> > Sungjin (James) Kim, PhD
> >
> > Postdoc, CCB in Harvard
> >
> > sungjinkim@fas.harvard.edu
> >
> > [Web] http://aspuru.chem.harvard.edu/james-sungjin-kim/
> >
> > [Linkedin] https://www.linkedin.com/in/jamessungjinkim
> >
> > [Facebook] https://www.facebook.com/jamessungjin.kim
> >
> > [alternative email] jamessungjin.kim@gmail.com
> >
> > —
> > You are receiving this because you commented.
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/tensorflow/tensorflow/issues/50#issuecomment-212488089>
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/50#issuecomment-212682339>
>
-- 

Sungjin (James) Kim, PhD

Postdoc, CCB in Harvard

sungjinkim@fas.harvard.edu

[Web] http://aspuru.chem.harvard.edu/james-sungjin-kim/

[Linkedin] https://www.linkedin.com/in/jamessungjinkim

[Facebook] https://www.facebook.com/jamessungjin.kim

[alternative email] jamessungjin.kim@gmail.com

**Comment by jjgh** [comment]
@nethsix judging from https://summerofcode.withgoogle.com/organizations/5131000665341952/ it seems that the TensorFlow Ruby API project has not been accepted in GsoC? Am I right?

**Comment by nethsix** [comment]
Yes that is correct. We had some applicants, and one of them worked with us
to submit a detailed plan but unfortunately also had some other internship
commitments which raised some concerns.

**Comment by riley-usagi** [comment]
That's so sad =(

**Comment by arkiver** [comment]
Hi @nethsix can we take a look at the detailed plan? Is it possible to make it available to us?

**Comment by nethsix** [comment]
Hey Akshay,

GSoC requires students to specify the deliverables each week for the entire
GSoC period, e.g., which API will be wrapped, created, etc. This enables
mentors to track progress accurately, and ensure students stick to their
commitments. I am not sure if it is permissible to share the plan wholesale
but if you think about it, this basically involves looking at Tensorflow
docs, tutorials, etc. to see what is the minimum amount of API to get
started, e.g., model a simple linear regression y = W.x + b requires
tf.placeholder, tf.Variable, tf.matmul, etc. and work from there. The
applicant has also created a repo to wrap the version API, i.e., return the
Tensorflow version, which may seem simple, but it acts a good foundation to
build from. Let me ask the student to share it, and maybe SciRuby can fork
it so that everyone can contribute from there. How does that sound? Any
suggestions?

khor

**Comment by jtoy** [comment]
I mentioned before, but I'd like to help push this forward by sponsoring part of it. We could put together a small fund or request to get this done. I would really like to see this down over the summer. Anyone interested in working on this?

-- Sent from my mobile

**Comment by nethsix** [comment]
Hey @jtoy. Let me ask SciRuby about sponsoring the program or would you
prefer to do the sponsorship directly?  We can also probably connect with
the student who applied for the GSoC.

**Comment by Arafatk** [comment]
@jtoy This seems like a wonderful opportunity and i would like to work on this.  I had applied to Gsoc this year for Supervised machine learning project in sciruby. I was close to being selected but the slots were limited so didn't make it. 

Anyways, I am very familiar with the code base of Sciruby projects such as nmatrix, Daru  and I have also made contributions to Sciruby and some other python projects too. Please, take a look at my application for Supervised machine learning gem [here](https://docs.google.com/document/d/1Ps58O_mXzO9rLtBA1RQGWNUdKcXU1wXOYN3Icqbs6kY/edit).

It appears to me that this is a challenging task, but if this could be made a sponsored project and a member of sciruby or Tensor flow could mentor this. Then I would be more than happy to work on this.

**Comment by Arafatk** [comment]
If you are cool with it. I could write an application for the project, similar to a gsoc application.

**Comment by jtoy** [comment]
yeah, an application is great, how much does Gsoc sponsor? I want to figure out how much this would cost and whether or not we need to find other sponsors.

**Comment by Arafatk** [comment]
@jtoy  In Gsoc a total stipend of 5500 dollars is given to the student and 500 dollars to the mentoring organization.  
As for this project, I think the sciruby community would be very welcoming to this idea. And if some mentors are free they could even provide some valuable insights for this project.

**Comment by nethsix** [comment]
Hey Arafat, @jtoy,

Arafat, I was talking to Sameer (one of the SciRuby admin) about this
earlier and he told me that he has reached out to you.

@jtoy, there is another student who submitted a GSoC proposal for
Tensorflow Ruby API. I've reached out to him as well.

I spoke to Sameer about @jtoy's sponsorship, and basically it may be best
for @jtoy to decide on the deliverables and deadlines, etc. SciRuby would
be more than happy to provide technical assistance required.

**Comment by riley-usagi** [comment]
Tensorflow Ruby API is very important for a large number of people from around the world (for example I'm from Russia), so maybe a good option would be to start a sort of open-raise funds for participants who wanted to undertake the development of this project, but so that the source was discovered during development, so that everyone can make a contribution.

**Comment by ankurayadav** [comment]
Hi Everyone,
@nethsix  informed me about this opportunity.
I think it would be great to develop this project since so many people are interested in this project.

I am sharing my GSoC project proposal please give me your valuable suggestions.

[proposal](https://docs.google.com/document/d/1_ysysHN5TWyXpAhsVNX1lGVGW7TF8tJhChjCzMY8nNY/edit?usp=sharing)

**Comment by v0dro** [comment]
Hi @jtoy and all potential sponsors:

I'm Sameer from SciRuby. I am this year's GSOC admin. After some discussion we have decided that we would be happy to provide any technical assistance to you and the student should you decide to take this project up.

I will send you a detailed email soon elaborating on how we can proceed forward with this issue, since a GitHub issue is really not the best place to discuss something like this.

In the meantime, please join the [SciRuby mailing list](https://groups.google.com/forum/#!forum/sciruby-dev) to get a heads up on the current progress of projects and our work methodology in general.

**Comment by arkiver** [comment]
@v0dro I think that's a great idea. Besides the GSoC students, a lot of other people are willing to contribute to this project too. This will help us get a direction on how to proceed and catch up with the progress.

**Comment by carlocasorzo** [comment]
+1

**Comment by jjgh** [comment]
Guys, thanks a lot for working on this; is it possible to keep us updated on the progress, here or anywhere else? :+1:

**Comment by v0dro** [comment]
Yes we'll be posting an official update soon.

**Comment by v0dro** [comment]
Hello,

Jason (@jtoy on GitHub) has volunteered to sponsor a student to write a Ruby TensorFlow wrapper over 3 months this summer. SciRuby will be providing technical assistance in this endeavor and

People interested in winning this grant should go over this blog entry for further details: http://www.somatic.io/blog/tensorflow-is-coming-to-ruby

Applicants are expected to provide a detailed proposal about the approach they will employ for writing the Ruby wrappers and provide monthly timeline over the next 3 months starting 1st of June. Deliverables should be on a monthly basis. We expect a weekly time line and proper evaluation of the approaches that you think are the best for this project.

Proposals will be accepted till 27 May. Please make sure you send us your proposals before this date. 

The sponsorship amount will be paid in monthly installments subject to completion of work for that particular month. The payment will be delayed if the applicant cannot deliver the promised material.

We also looking for additional sponsors for this project so interested people can get in touch by mailing me at sameer.deshmukh93@gmail.com.

Do spread the word!

**Comment by seanfcarroll** [comment]
It will be great to see this, the Machine Learning challenge site,
crowdai.org, is written in Ruby :-)

**Comment by yildizib** [comment]
+1

**Comment by JokerCatz** [comment]
+1 , and if TensorFlow not support Go, Java, Lua, Javascript, or R. ?

**Comment by chrhansen** [comment]
I'll also be willing to work on a Ruby-wrapper for Tensorflow!

**Comment by v0dro** [comment]
Get in touch with @Arafatk

**Comment by Arafatk** [comment]
@chrhansen That's nice. Please email me at arafat.da.khan@gmail.com and then we could discuss further.

**Comment by Arafatk** [comment]
I have been working on the Ruby API for Tensorflow and I have some interesting findings that I would like to share. The tensorflow.rb gem can be found on [somaticio/tensorflow.rb](https://github.com/somaticio/tensorflow.rb).
Aside from that, I have written three blog posts where I have given a very brief summary of the work
 
1. [Introductory blog post](https://medium.com/@Arafat./introducing-tensorflow-ruby-api-e77a477ff16e#.mhvj9ojlj)
2. [Developers blog post](https://medium.com/@Arafat./ruby-tensorflow-for-developers-2ec56b8668c5#.97tng1qqi)
3. [Image Recognition Tutorial](https://medium.com/@Arafat./image-recognition-in-ruby-tensorflow-df5d5c05389b#.ty1vygtrg)

I encourage everyone to read the blog posts and then install and play with Tensorflow.rb.  Any comments/suggestions are very welcome and you can always post your thoughts on the [gitter channel](https://gitter.im/tensorflowrb/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge). 

Special Thanks to Jason [Toy](url) ([Somatic.io](http://www.somatic.io/)), [Soon hin khor](https://github.com/nethsix) and [Sameer Deshmukh](https://github.com/v0dro) for mentoring and supporting me throughout the project.

**Comment by asimshankar** [comment]
Thanks @Arafatk ! Nice to hear.

From a cursory look, it does seem that this is building ontop of [c_api.h](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/c/c_api.h) using the approach outlined in the [how-to](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/how_tos/language_bindings/index.md), which is great.

If you have concerns around the C API or the how-to, please do let us know with a separate issue (since those files are meant to be useful across all language bindings).

**Comment by chrhansen** [comment]
To add to @Arafatk's comment above, if there are any Rubyers out there with extra time, feel free to pick up any outstanding issues from our Milestone 0.1.0: https://github.com/somaticio/tensorflow.rb/issues?q=is%3Aopen+is%3Aissue+milestone%3A0.1.0. Some of the issues don't require much knowledge of Tensorflow, but it's a good way to start learning about the inner workings of Tensorflow.

**Comment by johndpope** [comment]
fyi - I built this script to spit out ruby files (en masse) from tensor flow proto buffers for use with grpc https://gist.github.com/johndpope/5d176f4eebeb7ec983fa77d945c18fb1

If you want to discuss further -> jump on #grpc channel
https://tensor-flow-talk-invite.herokuapp.com/invite

**Comment by nethsix** [comment]
Neat. Thanks for sharing!

**Comment by jjgh** [comment]
@johndpope I think it should just be https://tensor-flow-talk-invite.herokuapp.com/ sorry for nitpicking :)

**Comment by v0dro** [comment]
Hey you guys should join the SciRuby slack channel too.

Here's the link to join: https://sciruby-slack.herokuapp.com/

**Comment by v0dro** [comment]
@Arafatk how's the Ruby Asso. Grant work coming along? Can you post your objectives for the grant period here so everyone has a clear idea of what you'll be doing?

**Comment by martinwicke** [comment]
I will close this bug, since it is not something that would live in the core tensorflow repository. If someone makes a Ruby API we'd be happy to link to it from tensorflow.org.

**Comment by Arafatk** [comment]
@martinwicke I am the lead developer of [Tensorflow.rb](https://github
.com/somaticio/tensorflow.rb) which is the Ruby API for Tensorflow.
I have seen a lot of support from the Ruby community for developing this. I
have been funded by @jtoy  [somaticio](http://www.somatic.io/)  and also
received the [Ruby grant](http://www.ruby.or.jp/en/news/20170427).
I have also contacted @asimshankar and @jhseu who have been very helpful
with my queries related to the issues that I faced in the development of
tensorflow.rb.
If you can link [tensorflow.rb](https://github.com/somaticio/tensorflow.rb)
from tensorflow.org that would be cool.

Thanks

**Comment by Arafatk** [comment]
@martinwicke I am the lead developer of [Tensorflow.rb](https://github.com/somaticio/tensorflow.rb) which is the Ruby API for Tensorflow.
I have seen a lot of support from the Ruby community for developing this and I have been funded by @jtoy  [somaticio](http://www.somatic.io/)  and also received the [Ruby grant](http://www.ruby.or.jp/en/news/20170427).    
I have also contacted @asimshankar and @jhseu who have been very helpful with my queries related to the issues that I faced in the development of tensorflow.rb.    
If you can link [tensorflow.rb](https://github.com/somaticio/tensorflow.rb) from tensorflow.org that would be cool.

Thanks

**Comment by asimshankar** [comment]
@Arafatk : it's already linked to from https://www.tensorflow.org/api_docs/ :)

**Comment by Arafatk** [comment]
@asimshankar Thanks you so much. This is great.

**Comment by LemonAndroid** [comment]
> @asimshankar Thanks you so much. This is great.

what is great?

and why?

**Comment by LemonAndroid** [comment]
and what is "is" 🗡

**Comment by Arafatk** [comment]
@LemonAndroid My repo https://github.com/somaticio/tensorflow.rb has been linked from the official docs.   
That's great.

**Comment by LemonAndroid** [comment]
@Arafatk oh, awesome, legoooooo

**Comment by LemonAndroid** [comment]
@Arafatk https://github.com/artyomche9/object_detection/blob/43a98307ab7f19861455acb2ee1441409f5636b7/app/models/image.rb @artyomche9
